### DISCUSSION TITLE
Make file picker asynchronous

### DISCUSSION THREAD

**Discussion by IvanVnucec** [helix-editor/helix#7489] [Ideas]
The problem is when File picker is opened on large mono-repos, it takes few seconds (too long) to open it. 
One solution would be to open File picker immediately and display empty window with some kind of message that the files are being loaded. When files are loaded, then we would display it.
In this way, user could open File picker and start to type immediately and the results would shown once they are loaded.
The feature is similar how the vscode does searching.

Also, this feature could be implemented also in the Global search which blocks once it has found the results. There is an issue related to this [here](https://github.com/helix-editor/helix/issues/5871).

**Comment by banocean** [comment]
For me this sounds more like an issue + suggested fix, than idea
